File: src/app/fibonacci/fibonacci.c
#include <emscripten.h>

EMSCRIPTEN_KEEPALIVE
int fibonacciLoop(int num)
{
  int a = 1;
  int b = 1;

  while (num-- > 1)
  {
    int t = a;
    a = b;
    b += t;
  }
  return b;
}

EMSCRIPTEN_KEEPALIVE
int fibonacciRec(int num)
{
  if (num <= 1)
    return 1;

  return fibonacciRec(num - 1) + fibonacciRec(num - 2);
}

File: src/app/fibonacci/fibonacci.ts
export const fibonacciLoop = (num: number) => {
  let a = 1;
  let b = 0;
  let temp;

  while (num >= 0) {
    temp = a;
    a = a + b;
    b = temp;
    num--;
  }

  return b;
};

export const fibonacciRec = (num: number) => {
  if (num <= 1) {
    return 1;
  }

  return fibonacciRec(num - 1) + fibonacciRec(num - 2);
};
